test/test_domain_utils.jl

```diff
@@ -0,0 +1,143 @@
+using HerbGrammar
+@testset verbose=true "Domain Utils" begin
```

Comment:
Would be cool if you could add some description to your tests. The tests themselves are hard to read and make sense of, especially when you are "new" to constraint solving/HerbConstraints. Give some intuition on what you are testing here.